infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerAdapter.java

```diff
@@ -29,34 +31,47 @@ public interface CorfuReplicationClusterManagerAdapter {
 
     /**
      * Query the topology information.
-     * @param useCached if it is true, used the cached topology, otherwise do a query to get the most
-     *                  recent topology from the real Cluster Manager.
+     * @param useCached if it is true, use the cached topology, otherwise do a query to get the most
+     *                  recent topology from the Cluster Manager/Topology Provider.
      * @return
      */
     TopologyConfigurationMsg queryTopologyConfig(boolean useCached);
 
-    // This is called when get a notification of cluster config change.
-    void updateTopologyConfig(TopologyConfigurationMsg newClusterConfigMsg);
+    /**
+     * Callback to update topology on cluster changes
+     */
+    void updateTopologyConfig(TopologyConfigurationMsg newClusterConfig);
 
-    // start to talk to the upper layer to get cluster topology information
+    /**
+     * Start cluster discovery against external topology provider
+     */
     void start();
 
-    // Stop the ClusterManager service
+    /**
+     * Shutdown cluster manager
+     */
     void shutdown();
 
     /**
-     * This API is used to query the log replication status when it is preparing a role type flip and
-     * the replicated tables should be in read-only mode.
+     * Query replication status for all ongoing sessions on source.
+     * This API is primarily used for UI display of metadata or in preparation for role switchover.
      *
-     * @return
+     * @return map of sessions to replication status
+     */
+    Map<LogReplicationSession, ReplicationStatus> queryReplicationStatus();
+
+    /**
+     * Query replication status on sink, currently only relevant status on sink is the consistency flag
+     *  on sink
      */
-    Map<String, LogReplicationMetadata.ReplicationStatusVal> queryReplicationStatus();
+    Map<LogReplicationSession, SinkReplicationStatus> queryStatusOnSink();
```

Comment:
I am not able to see why this would be a nice to have...
queryReplicationStatus() would be sufficient I think.

But I may not be seeing some scenarios. Open to discuss.

Reply:
Actually this is old code, at the beginning I had Source and Sink status separate, then I realized it would be better to make ReplicationStatus oneOf any of these two types, in that sense we could preserve just one interface and no need to separate these. 

@pankti-m we can get rid of this API and its caller the method isDataConsistent()